ffcx/codegeneration/integral_generator.py

```diff
@@ -446,17 +446,18 @@ def generate_block_parts(self, quadrature_rule: QuadratureRule, blockmap: Tuple,
         block_rank = len(blockmap)
         blockdims = tuple(len(dofmap) for dofmap in blockmap)
 
-        iq = self.backend.symbols.quadrature_loop_index
-
-        # Override dof index with quadrature loop index for arguments
-        # with quadrature element, to index B like B[iq*num_dofs + iq]
-        arg_indices = tuple(self.backend.symbols.argument_loop_index(i) for i in range(block_rank))
-        B_indices = []
-        for i in range(block_rank):
-            B_indices.append(arg_indices[i])
-        B_indices = list(B_indices)
+        iq_symbol = self.backend.symbols.quadrature_loop_index
+        iq = create_quadrature_index(quadrature_rule, iq_symbol)
 
         for blockdata in blocklist:
+
+            B_indices = []
+            args = [L.Symbol(name, L.DataType.INT) for name in ["i", "j", "k", "l"]]
```

Comment:
This already exists in `symbols.py` as `argument_loop_index()` - should we use (or delete from `symbols.py`)?